### ISSUE TITLE
Reintroduce scrolling within room list categories

### ISSUE THREAD

**Issue by swedneck** [vector-im/riot-web#14538]
**Is your suggestion related to a problem? Please describe.**
Right now if i want to see more of my DMs, i have to either expand the full category (which means i then have to scroll aaaaalllll the way down to re-collapse it, and then manually drag it up so it only shows 2 DMs again) or tediously drag to show more DMs (and then drag it back up again).

**Describe the solution you'd like**
It would be lovely to have the old category scrolling as an option (perhaps even per-category), as i found that infinitely more usable than the current situation.

**Comment by Biep** [comment]
Having the DM list expanded also means one cannot see all rooms with unread messages at once - because some are at the top of the DM list and some below the bottom, in the room list.
(The distinction between those sections is less useful now anyway, since one can no longer control it.  It now exposes some internal distinction rather than user categories.)

**Comment by SimmyD** [comment]
I prefer it the way it was.  :) scrolling within the category was alot more intuitive than this.

**Comment by t3chguy** [comment]
keyword: "letterboxes"

**Comment by Biep** [comment]
> keyword: "letterboxes"

Could you elaborate?  Is that a technical term?

**Comment by t3chguy** [comment]
It'd be a regression if it was accidental
Letterboxes was the internal term used for the old behaviour, just trying to make this issue more visible

**Comment by Biep** [comment]
OK, I see.  Sounds like a great term for rhythmic shouting at a protest march: "Let-ter-BOX!  Let-ter-BOX!"
What would be the rationale for removing a feature like that?  It wasn't "sitting in the way" of anything else, or making stuff complicated or cluttered, was it?

**Comment by t3chguy** [comment]
Issues like
https://github.com/vector-im/riot-web/issues/8042
https://github.com/vector-im/riot-web/issues/13521

**Comment by Biep** [comment]
#13521 would be a better option than what Element has - at least it would bring all rooms with unread messages to one place.
Part of #8042 seems to be about lost space - but "Show 60 more" is also lost space.
Having (advanced) options would be the obvious solution - a simple system with few evident options for casual users, but lots of advanced options allowing fine-tuning for whoever wants it.  In this case the code already exists, so it shouldn't be too hard to have all three versions implemented.

And in the longer term - it seems the issue has been misparameterised.  Reading the various issues I get the strong feeling there would be a quite different way to present those rooms, that would allow people to get what they want.  Can't get a handle on that yet, though..

**Comment by t3chguy** [comment]
> In this case the code already exists, so it shouldn't be too hard to have all three versions implemented.

No, it doesn't.

The new room list piggy-backs of a total rewrite to the underlying stores for efficiency and things like Message Previews. Bringing the letterbox style to the new backing stores would be non-trivial.

**Comment by Biep** [comment]
Oh, that is a real pity, because they serve a definite need - for some of us, at least.
And what about having the "one big list" option?  That would already make things much better too.

**Comment by t3chguy** [comment]
I'm not saying its an impossible thing to bring back as an option, just saying its not as easy as checking a box

**Comment by t3chguy** [comment]
> But the issue is about presentation - that should be fully separate from functionality, shouldn't it?

Yes, but the old "letterbox" Room List was built around the old stores, it doesn't understand the new stores

**Comment by Biep** [comment]
I appreciate that.

**Comment by 532910** [comment]
This issue is about usability that can be measured by the number of steps required to achieve the same result.
While before only cheap scrolling and 0 clicks was required, now I need to collapse/expand it each time.

**Comment by Cerothen** [comment]
I am also in agreement with the folks above. The letterbox method was significantly easier to use and quicker to obtain the desired outcome.. With the new larger icons and the loss of a quick scroll the accessibility is not as good as it was previously.

**Comment by altsalt** [comment]
After encountering this functionality, I brainstormed a bit in the Element Web/Desktop room and wanted capture it here.

### Problem:
Expanding a queue makes other queues (Favorites, People, etc.) inaccessible when there are more than a few rooms.

### Potential solutions:
1) Auto-collapse after clicking on a room
2) Change functionality to "show next/prev ##"
3) Collapse room list on queue collapse
4) Auto-collapse after a timer expires (my least favorite)

### Relevant feature request, that is potentially a semi-solution:
Allow sub-queues, such as the grouping of certain people into a subject matter list.

**Comment by ZerataX** [comment]
> Allow sub-queues, such as the grouping of certain people into a subject matter list.

maybe i misunderstand, but isn't that what communities are for. really hoping for that communities/groups overhaul

**Comment by altsalt** [comment]
> isn't that what communities are for

Potentially, but Communities are more like filters, and cross queue types. This proposal would be sub-queues within a particular queue.

**Comment by Biep** [comment]
But we'd definitely want to be able to use communities as tags (no need for two concepts there), and have queues for communities - and letterboxes for those queues.

**Comment by Cerothen** [comment]
> After encountering this functionality, I brainstormed a bit in the Element Web/Desktop room and wanted capture it here.
> ### Problem:
> 
> Expanding a queue makes other queues (Favorites, People, etc.) inaccessible when there are more than a few rooms.
> ### Potential solutions:
> 
>     1. Auto-collapse after clicking on a room
> 
>     2. Change functionality to "show next/prev ##"
> 
>     3. Collapse room list on queue collapse
> 
>     4. Auto-collapse after a timer expires (my least favorite)
> 
> 
> ### Relevant feature request, that is potentially a semi-solution:
> 
> Allow sub-queues, such as the grouping of certain people into a subject matter list.

Letter boxes was more functional than these options. Expanding a group adds a click to get to a room and if you want to access more than one room down the list then option 1 doesn't work (which was the only decent choice presented).

**Comment by twouters** [comment]
Can this issue be prioritized a bit? I'm getting pseudo anxiety attacks every time I've clicked show more instead of using the filter to see more than the first 5 entries.

**Comment by dragetd** [comment]
I do want to raise the question of what the actual issue is. As I interpret the messages in this thread, it is about not seeing room updates or notifications, loosing channels out of sight that one wants to see.

This is a valid concert and usability issue, I agree on that. But I disagree on re-introducing the scrollbars on each category being the best solution for this. Personally, I like it very much to have just one scrollbar. Did you know you could sort the rooms by 'last activity'? Personally I know more than once person who was surprised when I showed them this feature.

Some of the ideas here included to automatically move rooms with any activity to the very top. Or more automatic collapsing of categories based on some rules. For me personally it would already help if the rooms list was designed more sleek, i.e. less padding and smaller icons to basically double the amount of visible rooms. The category divider themself take quite a bit of space.
I would very much prefer that such options are also explored instead of 'just reverting to the old behavior', which for example I really did not like and prefer the new scrollbars, a lot.

**Comment by swedneck** [comment]
@dragetd as I said in the issue description this should be an option, so that those (like you) who don't like it can simply not enable it. I see no reason to start brainstorming fancy ideas when the old method was fine and could just be re-implemented as an option.

**Comment by altsalt** [comment]
The issue as I see it is that for folks with 100+ rooms in a single category, you will have more than the limited number of collapsed visible which are active. When you click to show more, it becomes a real hassle to see anything happening in other categories.

**Comment by Biep** [comment]
Letterboxes are very useful, as long as people can move rooms between categories.  I want active public rooms to come in sight, but without crowding out the less active personal discussions.
Ideally people can set up the tree for rooms themselves, and people who want it flat can then trivially have that.  If I want letterboxes for "private" and "work", I can do so, and see the activity in both at all times.
Dividers could - and in my view ought to be - indeed be a lot smaller.  Maybe they themselves should be collapsible/expandable.  Especially once letterboxes-in-letterboxes become a thing (wishing), small dividers (or a directory tree) would become an urgent necessity.

**Comment by Rafaeltheraven** [comment]
Now that spaces are becoming a thing, I'm finally moving to Element since Riot 1.6.4 is proving too old. The new interface is still, imo, terrible when you have a lot of rooms (as you automatically do when using bridges). It would be greatly appreciated if letterboxes could be reintroduced, if only as an optional labs setting,

**Comment by Cerothen** [comment]
This subject is pretty quiet for a long time here probably because everyone feels it wont be getting fixed.

The new method is inconvinient and slower to use. If I have a conversation that is just off the bottom of the number of bubbles shown for a category then I have to expand it or open the search and seach for it (which agrivatingly widens the user panel without restoring it...), I then have to manually compress the list again after getting the chat that I want, which means scrolling past 100s of conversations. Sure I could make a whole pile of spaces to try and cut my conversations up into categories or groups but that just seems like extra steps if I am actively talking to a pile of people between each space (and going back to "Home" has the aforementioned issue).

I would love to take advantage of the newer features like spaces from the web client but am stuck using the last release of Riot before it became Element since the changes to the userlist make the client unusable.

Why the letterbox feature was removed I will never quite understand and I would love to see the arguemnt that says the new way is any better.

**Comment by Biep** [comment]
We wrote that we wanted it back, and why, and when no good counterarguments were given the discussion fell flat.  But we are still waiting, and deploring the lack of letterboxes on a daily basis.

**Comment by t3chguy** [comment]
The design team opted for a single scrollable list instead
